sequencers/modeshape-sequencer-video/src/main/java/org/modeshape/sequencer/video/VideoMetadata.java

```
@@ -0,0 +1,200 @@
+                }
+                if (key.toLowerCase().equals("encoder")) {
+                    encoder = metadata.getValue(key);
+                }
+            }
+
+            if (container.getDuration() != Global.NO_PTS) {
+                duration = (float) container.getDuration() / 1000.0 / 1000.0; // convert to seconds
+            }
+            if (container.getBitRate() > 0) {
+                bitrate = container.getBitRate();
+            }
+
+
+            for (int i = 0; i < container.getNumStreams(); i++) {
+                    StreamMetadata streamMetadata = new StreamMetadata();
+                    DemuxerStream stream = container.getStream(i);
+
+                    Decoder coder = stream.getDecoder();
+
+                    if (coder.getCodecType() == Type.MEDIA_AUDIO) {
+                        streamMetadata.setStreamType("audio");
+                    } else if (coder.getCodecType() == Type.MEDIA_VIDEO) {
+                        streamMetadata.setStreamType("video");
+                    } else {
+                        streamMetadata.setStreamType("unknown");
+                    }
+
+                    if (coder.getCodec() != null) {
+                        streamMetadata.setCodec(coder.getCodec().getName());
+                    }
+                    streamMetadata.setSamplerate(nullValueIfInvalid(coder.getSampleRate()));
+                    streamMetadata.setChannels(nullValueIfInvalid(coder.getChannels()));
+                    streamMetadata.setWidth(nullValueIfInvalid(coder.getWidth()));
+                    streamMetadata.setHeight(nullValueIfInvalid(coder.getHeight()));
+                    if (coder.getTimeBase() != null) {
+                        streamMetadata.setFramerate(nullValueIfInvalid(1.0 / coder.getTimeBase().getDouble()));
+                    }
+                    streams.add(streamMetadata);
+                }
+            return true;
+        } catch (Exception e) {
+            e.printStackTrace();
```

this should be logged or bubbled up, instead of calling `printStackTrace`